### PR TITLE
Improve various code styles

### PR DESCRIPTION
**Pull request by lamby**
This PR includes a large number of code refactoring within the backend (Django) part of the application.

### CODE REVIEW

**Comment by lamby** [comment]
@isms @reubano  Any ideas on this testsuite failure? Can't seem to work out what is actually really broken, let alone how I could have done it...

**Comment by reubano** [comment]
Possibly related to https://github.com/concept-to-clinic/concept-to-clinic/pull/305#issuecomment-363400089

**Comment by isms** [comment]
@reubano Can you dig in and find out what the source is?

@lamby I say we let that tests fail for a while while Reuben looks into it so you're not blocked :)

**Comment by lamby** [comment]
As in, I can't *quite* understand the error message!

**Comment by reubano** [comment]
> @reubano Can you dig in and find out what the source is?

@isms @lamby can either of you verify that bug?

**Comment by lamby** [comment]
@isms  I'm not blocked, more that I wouldn't want to be *introducing* any new issue. Indeed, this PR can be merged (renaming to match)

File: interface/backend/api/urls.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from collections import OrderedDict
+import collections
```
Comment (WGierke) [inline interface/backend/api/urls.py]: Why do you prefer to import the whole module here while you changed in 5302cfe401a1f77fa878fd8ad1bc26dd2d77e224 to directly import methods from a module?

Comment (lamby) [reply interface/backend/api/urls.py]: So in general I prefer to import the modules that Python ships with ("batteries included") as top-level modules. However, I am not 100% dogmatic on this (!) and make some exceptions if that's going to make the code unspeakably ugly. For example,, imagine trying to parse:
```
os.path.join(os.path.join(os.path.join(os.path.ditpath(os.path.join(os.path.join(os.path.join(....)))
```
Make sense? :)

**Comment by lamby** [comment]
Any luck all? Would love to get this merged! :)

File: interface/backend/api/urls.py (hunk 1 above)
Comment (WGierke) [reply interface/backend/api/urls.py]: Yes, thank you :)

**Pull request merged**